url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/data/FixedMatrix4x4_64F.java

```diff
@@ -0,0 +1,208 @@
+    public double a41,a42,a43,a44;
+
+    public FixedMatrix4x4_64F() {
+    }
+
+    public FixedMatrix4x4_64F( double a11,double a12,double a13,double a14,
+                              double a21,double a22,double a23,double a24,
+                              double a31,double a32,double a33,double a34,
+                              double a41,double a42,double a43,double a44)
+    {
+        this.a11 = a11;
+        this.a12 = a12;
+        this.a13 = a13;
+        this.a14 = a14;
+        this.a21 = a21;
+        this.a22 = a22;
+        this.a23 = a23;
+        this.a24 = a24;
+        this.a31 = a31;
+        this.a32 = a32;
+        this.a33 = a33;
+        this.a34 = a34;
+        this.a41 = a41;
+        this.a42 = a42;
+        this.a43 = a43;
+        this.a44 = a44;
+    }
+
+    public FixedMatrix4x4_64F( FixedMatrix4x4_64F o ) {
+        this.a11 = o.a11;
+        this.a12 = o.a12;
+        this.a13 = o.a13;
+        this.a14 = o.a14;
+        this.a21 = o.a21;
+        this.a22 = o.a22;
+        this.a23 = o.a23;
+        this.a24 = o.a24;
+        this.a31 = o.a31;
+        this.a32 = o.a32;
+        this.a33 = o.a33;
+        this.a34 = o.a34;
+        this.a41 = o.a41;
+        this.a42 = o.a42;
+        this.a43 = o.a43;
+        this.a44 = o.a44;
+    }
+
+    @Override
+    public double get(int row, int col) {
+        return unsafe_get(row,col);
+    }
+
+    @Override
+    public double unsafe_get(int row, int col) {
+        if( row == 0 ) {
+            if( col == 0 ) {
+                return a11;
+            } else if( col == 1 ) {
+                return a12;
+            } else if( col == 2 ) {
+                return a13;
+            } else if( col == 3 ) {
+                return a14;
+            }
+        } else if( row == 1 ) {
+            if( col == 0 ) {
+                return a21;
+            } else if( col == 1 ) {
+                return a22;
+            } else if( col == 2 ) {
+                return a23;
+            } else if( col == 3 ) {
+                return a24;
+            }
+        } else if( row == 2 ) {
+            if( col == 0 ) {
+                return a31;
+            } else if( col == 1 ) {
+                return a32;
+            } else if( col == 2 ) {
+                return a33;
+            } else if( col == 3 ) {
+                return a34;
+            }
+        } else if( row == 3 ) {
+            if( col == 0 ) {
+                return a41;
+            } else if( col == 1 ) {
+                return a42;
+            } else if( col == 2 ) {
+                return a43;
+            } else if( col == 3 ) {
+                return a44;
+            }
+        }
+        throw new IllegalArgumentException("Row and/or column out of range. "+row+" "+col);
+    }
+
+    @Override
+    public void set(int row, int col, double val) {
+        unsafe_set(row,col,val);
+    }
+
+    @Override
+    public void unsafe_set(int row, int col, double val) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 21. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.